docs/pages/guides/local-app-development.mdx

```diff
@@ -0,0 +1,79 @@
+---
+title: Local app development
+description: Learn how to compile and build your app locally when using Expo.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { BoxLink } from '~/ui/components/BoxLink';
+import { BookOpen02Icon, BuildIcon } from '@expo/styleguide-icons';
+import { CODE } from '~/ui/components/Text';
+
+To build your project into an app locally using your machine, you have to manually generate native code before testing the debug build or creating a production build for it to submit to the app store. There are two ways you can build your app locally. This guide provides a brief introduction to both methods and references to other guides that are necessary to create this workflow.
+
+## Prerequisites
+
+To follow the steps described in this guide, install the following tools on your local machine:
+
+- [Android Studio](/workflow/android-studio-emulator/) to build for Android
+- [Xcode](/workflow/ios-simulator/) on macOS to build for iOS
+
+## Local app compilation
+
+To build your project locally you can use compile commands from Expo CLI which generates the **android** and **ios** directories:
+
+<Terminal
+  cmd={[
+    '# Build native Android project',
+    '$ npx expo run:android',
+    '# Build native iOS project',
+    '$ npx expo run:iOS',
+  ]}
+/>
+
+The above commands compile your project into an app locally. You can also add the `--device` flag to create a debug build of your project and run it on a connected device. These compilation commands initially run `npx expo prebuild` to generate native directories (**android** and **ios**) before building.
+
+After building your project for the first time, you can modify your project's configuration or native code. However, it is recommended to make these modifications using [app config](/workflow/configuration/) or by creating a [config plugin](/config-plugins/introduction/).
+
+To learn more about how compilation and prebuild works, see the following guides:
```

Comment:
we might want to explicitly call out gitignoring ios and android directories and `prebuild --clean` here, since not knowing about these concepts trip people up pretty quickly once they start doing local builds

Reply:
Added info about the importance of .gitignore and `--clean` flag:

![CleanShot 2023-09-18 at 22 16 35](https://github.com/expo/expo/assets/10234615/3054dae3-cd24-4160-bd9d-8f791e96639c)